qiita_db/sql_connection.py

```diff
@@ -1047,10 +1067,54 @@ def rollback(self):
         except Exception:
             self._connection.close()
             raise
+        # Execute the post rollback functions
+        self._funcs_executor(self._post_rollback_funcs, "rollback")
 
     @property
     def index(self):
         return len(self._queries) + len(self._results)
 
+    @_checker
+    def add_post_commit_func(self, func):
+        """Adds a post commit function
+
+        The function added will be executed after the next commit in the
+        transaction, unless a rollback is executed. This is useful, for
+        example, to perform some filesystem clean up once the transaction is
+        committed.
+
+        Parameters
+        ----------
+        func : function
+            The function to add for the post commit functions
+
+        Notes
+        -----
+        func should not accept any parameter, i.e. it should allow to be
```

Comment:
As stated above, this may need parameters. I can see this being useful for passing in a list of files or some other list of items that need to be worked on.